ledger/participant-integration-api/src/main/scala/platform/store/dao/events/ACSReader.scala

```diff
@@ -60,28 +62,38 @@ class FilterTableACSReader(
     val idQueryLimiter =
       new QueueBasedConcurrencyLimiter(idFetchingParallelism, executionContext)
 
+    val idQueryConfiguration = IdQueryConfiguration(
+      maxIdPageSize = idPageSize,
+      idPageWorkingMemoryBytes = idPageWorkingMemoryBytes,
+      filterSize = filters.size,
+      idPageBufferSize = idPageBufferSize,
+    )
+
     def toIdSource(filter: Filter): Source[Long, NotUsed] =
-      idSource(idPageBufferSize)(fromExclusive =>
-        idQueryLimiter.execute(
+      idSource(
+        idQueryConfiguration = idQueryConfiguration,
+        pageBufferSize = idPageBufferSize,
+      )(idQuery =>
+        idQueryLimiter.execute {
           dispatcher.executeSql(metrics.daml.index.db.getActiveContractIds) { connection =>
             val result = eventStorageBackend.activeContractEventIds(
               partyFilter = filter.party,
               templateIdFilter = filter.templateId,
-              startExclusive = fromExclusive,
+              startExclusive = idQuery.fromExclusiveEventSeqId,
               endInclusive = activeAt._2,
-              limit = idPageSize,
+              limit = idQuery.pageSize,
             )(connection)
             logger.debug(
               s"getActiveContractIds $filter returned #${result.size} ${result.lastOption
                   .map(last => s"until $last")
                   .getOrElse("")}"
             )
-            result
+            result.toArray
```

Comment:
I was thinking about introducing here a direct conversion to Array, but seemed like not worth the effort: what we would gain only is preventing the memory pressure of a short-lived temporary Vector object.
This is a concern here only because there is a multiplier on how much many of these we will store for the entire lifecycle for a stream (filter-size). In other cases it is usually a handful / defined by some buffer.
Many the time these vectors are temporary data, and getting transformed to something else (like here). Also wast majority of times we need something more than a primitive where the use of Arrays is diminishing.